integration/scheduler.go

```diff
@@ -9,19 +9,61 @@ import (
 	"github.com/kubernetes-sig-testing/frameworks/integration/internal"
 )
 
+// Scheduler is kube-scheduler. It can be used as an `AdditionalComponent` for
+// the control plane.
 type Scheduler struct {
-	URL          *url.URL
-	Path         string
-	Args         []string
+	// URL is the address the Scheduler should listen on for connections.
+	//
+	// If this is not specified, we default to a random free port on localhost.
+	URL *url.URL
+
+	// Path is the path to the scheduler binary.
+	//
+	// If this is left as the empty string, we will attempt to locate a binary,
+	// by checking for the TEST_ASSET_KUBE_SCHEDULER environment variable, and
+	// the default test assets directory. See the "Binaries" section above (in
+	// doc.go) for details.
+	Path string
+
+	// Args is a list of arguments which will passed to the Scheduler binary.
+	// Before they are passed on, they will be evaluated as go-template strings.
+	// This means you can use fields which are defined and exported on this
+	// Scheduler struct (e.g. "--master={{ .APIServerURL.String }}",
+	// Those templates will be evaluated after the defaulting of the Scheduler's
+	// fields has already happened and just before the binary actually gets
+	// started. Thus you have access to caluclated fields like `URL` and others.
+	//
+	// If not specified, the minimal set of arguments to run the Scheduler will
+	// be used.
+	Args []string
+
+	// APIServerURL is the URL pointing to the APIServer of the control plane
+	// this Scheduler should connect to.
+	//
+	// It can be set directly on the struct, or indirectly by calling the
+	// `RegisterTo` method.
+	// This setting is mandatory and an error will be emitted when this is not
+	// set at start time.
 	APIServerURL *url.URL
+
+	// StartTimeout, StopTimeout specify the time the APIServer is allowed to
+	// take when starting and stoppping before an error is emitted.
+	//
+	// If not specified, these default to 20 seconds.
 	StartTimeout time.Duration
 	StopTimeout  time.Duration
-	Out          io.Writer
-	Err          io.Writer
+
+	// Out, Err specify where APIServer should write its StdOut, StdErr to.
```

Comment:
not  apiserver

Reply:
Thanks. I will recheck on the docs and typos overall.